### PR TITLE
Scc 3699/cache source map

### PR DESCRIPTION
**Pull request by charmingduchess**
Eliminates axios calls per invocation of nypl source mapping functions by adding an instance() method which returns a cached instance.
Also removes discovery-store-models as a dependency. Are we planning to do anything with the soon to be retired DHI components (as in maybe delete them from github at some point down the line?)

### CODE REVIEW

File: lib/utils/nypl-source-mapper.js [hunk 1]
```diff
@@ -49,11 +44,27 @@ class NyplSourceMapper {
...
 }
 
+let sourceMapperInstance = null
+
+NyplSourceMapper.instance = async () => {
+  if (!sourceMapperInstance) {
+    const resp = await axios.get(`https://raw.githubusercontent.com/NYPL/nypl-core/${process.env.NYPL_CORE_VERSION || 'master'}/mappings/recap-discovery/nypl-source-mapping.json`)
+    const mapping = resp.data
+    sourceMapperInstance = new NyplSourceMapper(mapping)
+  }
```
[the remark this answers is not in the document]
Comment (charmingduchess) [reply lib/utils/nypl-source-mapper.js:61]: can we access `.data` on the yet to be resolved promise?

[the remark this answers is not in the document]
Comment (nonword) [reply lib/utils/nypl-source-mapper.js:61]: The function that you pass to a `.then` is fired only when the Promise resolves.

[the remark this answers is not in the document]
Comment (charmingduchess) [reply lib/utils/nypl-source-mapper.js:61]: ok. i think i understand the pattern now.

**Review by nonword — APPROVED** [review]
Looks good. One minor comment.

**Review by nonword — APPROVED** [review]

**Pull request merged**